pkg/transform/kubernetes/kubernetes.go

```diff
@@ -116,25 +117,37 @@ func createServices(app *spec.App) ([]runtime.Object, error) {
 	return svcs, nil
 }
 
-func createDeployment(app *spec.App) *ext_v1beta1.Deployment {
-	// bare minimum deployment
-	return &ext_v1beta1.Deployment{
-		ObjectMeta: api_v1.ObjectMeta{
-			Name:   app.Name,
-			Labels: app.Labels,
-		},
-		Spec: ext_v1beta1.DeploymentSpec{
-			Replicas: app.Replicas,
-			Template: api_v1.PodTemplateSpec{
-				ObjectMeta: api_v1.ObjectMeta{
-					Name:   app.Name,
-					Labels: app.Labels,
-				},
-				// get pod spec out of the original info
-				Spec: api_v1.PodSpec(app.PodSpec),
-			},
-		},
+func createDeployment(app *spec.App) (*ext_v1beta1.Deployment, error) {
+
+	// We are merging whole DeploymentSpect with PodSpec.
+	// This means that someone could specify containers in template.spec and also in top level PodSpec.
+	// This stupid check is supposed to make sure that only one of them set.
+	// TODO: merge DeploymentSpec.Template.Spec and top level PodSpec
+	if !(reflect.DeepEqual(app.DeploymentSpec.Template.Spec, api_v1.PodSpec{}) || reflect.DeepEqual(app.PodSpec, api_v1.PodSpec{})) {
+		return nil, fmt.Errorf("Pod can't be specfied in two places. Use top level PodSpec or template.spec (DeploymentSpec.Template.Spec) not both")
+	}
+
+	deployment := ext_v1beta1.Deployment{}
+
+	// use top level name and labels for deployment
+	deployment.ObjectMeta.Name = app.Name
+	deployment.ObjectMeta.Labels = app.Labels
```

Comment:
When we do the deep indentation, the code actually looks a bit the objects that will get generated, which is helpful while reading.
Also, everywhere we have done it with the indentation way.
And, the object to which we want to add additional logic, we create it separately and substitute it in the indentation.
Idk, I really like the indentation way :( a lot more readable.
@surajssd WDYT?